python/ray/tests/unit/test_resource_and_label_spec.py

```diff
@@ -0,0 +1,361 @@
+    monkeypatch.setenv("RAY_NODE_REGION", "us-west1")
+    monkeypatch.setenv("RAY_NODE_ZONE", "us-west1-a")
+
+    # Patch accelerator manager to return an accelerator type
+    mock_mgr = (
+        patch("ray._private.accelerators.get_accelerator_manager_for_resource")
+        .start()
+        .return_value
+    )
+    patch(
+        "ray._private.accelerators.get_all_accelerator_resource_names",
+        return_value=["GPU"],
+    ).start()
+    mock_mgr.get_current_node_num_accelerators.return_value = 1
+    mock_mgr.get_current_process_visible_accelerator_ids.return_value = ["0"]
+    mock_mgr.get_resource_name.return_value = "GPU"
+    mock_mgr.get_current_node_accelerator_type.return_value = "A100"
+    mock_mgr.get_current_node_additional_resources.return_value = {}
+
+    spec = ResourceAndLabelSpec()
+    spec.resolve(is_head=False)
+
+    # Verify all default labels are present
+    assert spec.labels.get("ray.io/node-group") == "worker-group-1"
+    assert spec.labels.get("ray.io/market-type") == "spot"
+    assert spec.labels.get("ray.io/availability-region") == "us-west1"
+    assert spec.labels.get("ray.io/availability-zone") == "us-west1-a"
+    assert spec.labels.get("ray.io/accelerator-type") == "A100"
+    assert spec.resolved()
+
+
+def test_resolve_raises_if_exceeds_visible_devices(monkeypatch):
+    """Check that ValueError is raised when requested accelerators exceed visible IDs."""
+    monkeypatch.setenv(ray_constants.LABELS_ENVIRONMENT_VARIABLE, "{}")
+
+    mock_mgr = (
+        patch("ray._private.accelerators.get_accelerator_manager_for_resource")
+        .start()
+        .return_value
+    )
+    patch(
+        "ray._private.accelerators.get_all_accelerator_resource_names",
+        return_value=["GPU"],
+    ).start()
+    mock_mgr.get_current_node_num_accelerators.return_value = 5
+    mock_mgr.get_current_process_visible_accelerator_ids.return_value = [
+        "0",
+        "1",
+    ]  # 2 visible IDs
+    mock_mgr.get_resource_name.return_value = "GPU"
+
+    spec = ResourceAndLabelSpec()
+    spec.num_gpus = 3  # request 3 GPUs
+
+    with pytest.raises(ValueError, match="Attempting to start raylet"):
+        spec.resolve(is_head=False)
+
+
+@patch("ray._private.usage.usage_lib.record_hardware_usage", lambda *_: None)
+def test_resolve_sets_accelerator_resources(monkeypatch):
+    """Verify that GPUs/TPU values are auto-detected and assigned properly."""
+    # Mock a node with GPUs with 4 visible IDs
+    mock_mgr = (
+        patch("ray._private.accelerators.get_accelerator_manager_for_resource")
+        .start()
+        .return_value
+    )
+    patch(
+        "ray._private.accelerators.get_all_accelerator_resource_names",
+        return_value=["GPU"],
+    ).start()
+    mock_mgr.get_current_node_num_accelerators.return_value = 4
+    mock_mgr.get_current_process_visible_accelerator_ids.return_value = [
+        "0",
+        "1",
+        "2",
+        "3",
+    ]
```

Comment:
Instead of mocking, can we define a FakeGPUAcceleratorManager that implements AcceleratorManager and return it from `get_accelerator_manager_for_resource`